specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/cadl/multivariate/models.cadl

```diff
@@ -0,0 +1,370 @@
+}
+
+@doc("An optional field, indicating the manner to align multiple variables.")
+model AlignPolicy {
+  @doc("""
+An optional field, indicating how to align different variables to the same
+time-range. Either Inner or Outer. 
+""")
+  alignMode?: AlignMode;
+
+  @doc("""
+An optional field, indicating how missing values will be filled. One of
+Previous, Subsequent, Linear, Zero, Fixed.
+""")
+  fillNAMethod?: FillNAMethod;
+
+  @doc("An optional field. Required when fillNAMethod is Fixed.")
+  paddingValue?: float32;
+}
+
+@doc("Diagnostics information to help inspect the states of model or variable.")
+model DiagnosticsInfo {
+  modelState?: ModelState;
+  variableStates?: VariableState[];
+}
+
+model ModelState {
+  @doc("""
+This indicates the number of passes of the entire training dataset the
+algorithm has completed.
+""")
+  epochIds?: int32[];
+
+  @doc("""
+List of metrics used to assess how the model fits the training data for each
+epoch.
+""")
+  trainLosses?: float32[];
+
+  @doc("""
+List of metrics used to assess how the model fits the validation set for each
+epoch.
+""")
+  validationLosses?: float32[];
+
+  @doc("Latency for each epoch. ")
+  latenciesInSeconds?: float32[];
+}
+
+@doc("Response of getting a model.")
+@resource("multivariate/models")
+model Model {
+  @doc("Model identifier.")
+  @key
+  @format("uuid")
+  modelId: string;
+
+  @doc("Date and time (UTC) when the model was created.")
+  createdTime: zonedDateTime;
+
+  @doc("Date and time (UTC) when the model was last updated.")
+  lastUpdatedTime: zonedDateTime;
+
+  @doc("""
+Training result of a model including its status, errors and diagnostics
+information.
+""")
+  modelInfo?: ModelInfo;
+}
+
+@doc("Response of listing models.")
+@pagedResult
+model ModelList {
+  @doc("List of models.")
+  @items
+  models: Model[];
+
+  @doc("Number of trained multivariate models.")
+  currentCount: int32;
+
+  @doc("Maximum number of models that can be trained for this Anomaly Detector resource.")
+  maxCount: int32;
+
+  @doc("The link to fetch more models.")
+  @nextLink
+  nextLink?: string;
+}
+
+model LastDetectionRequest {
+  @doc("""
+This contains the inference data, including the name, timestamps(ISO 8601) and
+values of variables.
+""")
+  variables: VariableValues[];
+
+  @doc("""
+An optional field, which is used to specify the number of top contributed
+variables for one anomalous timestamp in the response. The default number is
+10.
+""")
+  topContributorCount: int32;
+}
+
+model VariableValues {
+  @doc("Variable name of last detection request.")
+  variable: string;
+
+  @doc("Timestamps of last detection request")
+  timestamps: string[];
+
+  @doc("Values of variables.")
+  values: float32[];
+}
+
+model LastDetectionResult {
+  variableStates?: VariableState[];
+  results?: AnomalyState[];
+}
+
+@doc("Error response")
+@error
+model ResponseError {
+  @header
+  @doc("Error code.")
+  "x-ms-error-code"?: string,
+  ...ErrorResponse
+}
```

Comment:
As discussed in Arc Board: there are both error definition for `multivariate` and `univirate`, so it is better to add prefix to distinguish them more clearly.

Reply:
BTW the structure is wrong (not sure from history reason or just mistake).

The error response should have an "error" property, which then contains "code" and "message".

https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#handling-errors
and
https://github.com/Azure/cadl-azure/blob/main/packages/cadl-azure-core/lib/foundations.cadl#L65-L70

Reply:
Error updates are pending testing. The definitions may be updated after testing error scenarios in each language SDK.